gitlab/client.py

```diff
@@ -1348,7 +1450,9 @@ def execute(
 
         while True:
             try:
-                result = self._client.execute(parsed_document, *args, **kwargs)
+                result = await self._client.execute_async(
+                    parsed_document, *args, **kwargs
+                )
```

Comment:
Maybe I could put some of the error handling into the base client if you think it's worth it, this was just a quick start here.

Reply:
I'll put this into a helper so it's reused and we have a smaller diff to review here.